content/100-getting-started/03-prisma-postgres/115-import-from-mysql-database.mdx

```diff
@@ -0,0 +1,346 @@
+
+## 2. Connect directly to a Prisma Postgres instance
+
+In this step, you'll use a secure [TCP tunnel](/postgres/tcp-tunnel) to connect to your Prisma Postgres instance.
+
+You'll need the Prisma Postgres connection URL from [step 1](/getting-started/prisma-postgres/import-from-mysql-database#1-create-a-new-prisma-postgres-database):
+
+```no-copy
+prisma+postgres://accelerate.prisma-data.net/?api_key=ey...
+```
+
+Open your terminal and set the `DATABASE_URL` environment variable to the value of your Prisma Postgres database URL (replace the `__API_KEY__` placeholder with the API key of your actual database connection URL):
+
+```terminal
+export DATABASE_URL="prisma+postgres://accelerate.prisma-data.net/?api_key=__API_KEY__"
+```
+
+Next, start the TCP tunnel using the `@prisma/ppg-tunnel` package, by executing the following command:
+
+<CodeWithResult outputResultText="" expanded={true}>
+<cmd>
+
+```terminal
+npx @prisma/ppg-tunnel --host 127.0.0.1 --port 5433 
+```
+
+:::note
+
+You can specify a different port by providing your own host and port values using the `--port` and `--host` flags. Just be sure to use the same host and port values consistently throughout the guide.
+
+:::
+
+</cmd>
+
+<cmdResult>
+
+```code no-copy wrap
+Prisma Postgres auth proxy listening on 127.0.0.1:5433 🚀
+
+Your connection is authenticated using your Prisma Postgres API key.
+...
+
+==============================
+hostname:  127.0.0.1
+port:      5433
+username:  <anything>
+password:  <none>
+==============================
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+:::note
+
+*Keep your current terminal window or tab open* so that the tunnel process continues running and the connection remains open.
+
+:::
+
+## 3. Migrate your MySQL data to Prisma Postgres using pgloader
+
+Now that you have an active connection to your Prisma Postgres instance, you'll use pgloader to export data from your MySQL database to Prisma Postgres.
+
+Open a separate terminal window and create a `config.load` file:
+
+```terminal
+touch config.load
+```
+
+Open the `config.load` file in your preferred text editor and copy-paste the following configuration:
+
+```text file=config.load
+LOAD DATABASE
+    FROM mysql://username:password@host:PORT/database_name
+    INTO postgresql://user:password@127.0.0.1:5433/postgres
+
+WITH quote identifiers,    -- preserve table/column name case by quoting them
+     include drop,
+     create tables,
+     create indexes,
+     reset sequences
+
+ALTER SCHEMA 'database_name' RENAME TO 'public';
+```
+
+Make sure to update the following details in the `config.load` file:
+
+- `FROM URL`: Replace `username`, `password`, `host`, `PORT`, and `database_name` with the actual connection details of your MySQL database.
+- `INTO URL`: Update this with your TCP tunnel details if you’re using a custom `host` and `port` (in this example, it’s `127.0.0.1` and `port` `5433` for consistency).
```

Comment:
```suggestion
- `INTO URL`: If you’ve used a different value than `--port 5433` when starting the TCP tunnel above, you need to replace the `5433` with that value here. Otherwise you can leave it as is. (Note that `user` and `password` values will be ignored.)
```